### PR TITLE
Modifying

### PR DESCRIPTION
**Pull request by arielofer**
נראה לי שהקלאסים החדשים יעשו את העבודה
dexter

### CODE REVIEW

File: .pytest_cache/v/cache/nodeids [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (agamm) [inline .pytest_cache/v/cache/nodeids]: Put this directory in the .gitignore :)

File: .pytest_cache/v/cache/stepwise [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (agamm) [inline .pytest_cache/v/cache/stepwise]: Put this directory in the .gitignore :)

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (agamm) [inline .vscode/settings.json]: Also, .gitignore.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,6 @@
 # gamebook.py
 a python library for building text based choose your own adventure games
+
+לאגם:
+עוד לא הרצתי טסטים על הקלאסים החדשים אז אין לי מושג אם הם עובדים
```
Comment (agamm) [inline README.md]: Next time, add this comment in the PR itself.
Regarding the question. Upload some tests so we will know then 👍

File: temp_library/helpers.py [hunk 5]
```diff
@@ -47,36 +47,39 @@ def statsdsply(trait1, trait2, trait3):
...
+    dice_num = trait.getDiceNum()
+    Trait_value = trait.getValue()
+
+    for i in range(dice_num):
+        dice_roll += random.randint(1, 6)
 
     if arg == 'less':
-        if dice <= num:
+        if dice_roll <= Trait_value:
             print("")
```
Comment (agamm) [inline temp_library/helpers.py]: Return a value, don't print (so you could use a different presentation layer (like a website) later on).

File: temp_library/helpers.py [hunk 6]
```diff
@@ -47,36 +47,39 @@ def statsdsply(trait1, trait2, trait3):
...
-def roll(num, arg):
+def roll(trait, arg):
     # dice rolling function
     input("press enter to roll the dice...")
-    dice = 0
-    for i in range(1):
-        dice += random.randint(1, 6)
+    dice_roll = 0
+    dice_num = trait.getDiceNum()
+    Trait_value = trait.getValue()
```
Comment (agamm) [inline temp_library/helpers.py]: Capital case?

File: temp_library/classes.py [hunk 7]
```diff
@@ -42,7 +60,13 @@ def ShowKeys(self):
         return key_list
 
 
-# class ChanceOption(Option):
+class ChanceOption(Option):
+    def DecisionDraw(self):
+        wheights = []
+        for i in range(len(self.decision)):
+            wheights[i] = self.decision[i].getSuccessRate()
+        draw = random.choices(self.decision,wheights,k=1)
```
Comment (agamm) [inline temp_library/classes.py]: Cool!

File: temp_library/classes.py [hunk 8]
```diff
@@ -1,13 +1,30 @@
 import temp_library
```
Comment (agamm) [inline temp_library/classes.py]: Split every class to a different file

File: temp_library/classes.py [hunk 9]
```diff
@@ -1,13 +1,30 @@
 import temp_library
+import random
 
+class Decision(object):
```
Comment (agamm) [inline temp_library/classes.py]: Cool!

File: temp_library/classes.py [hunk 10]
```diff
@@ -1,13 +1,30 @@
...
+        return self.successRate
 
-# class Decision:
 
 class Scene(object):
-    def __init__(self, desc, choice, decision):
+    def __init__(self, name, desc, choice):
+        self.name = name
         self.desc = desc
         self.choice = choice
```
Comment (agamm) [inline temp_library/classes.py]: choices?

File: temp_library/classes.py [hunk 11]
```diff
@@ -1,13 +1,30 @@
...
-        self.decision = decision
+        
+
+    def GetName(self):
```
Comment (agamm) [inline temp_library/classes.py]: Try to use a python linter (python uses snake_case not CamelCase)

Comment (agamm) [reply temp_library/classes.py]: Search for pylint/black they help linting (fixing code styles).

Comment (agamm) [reply temp_library/classes.py]: or flake8

File: temp_library/classes.py [hunk 12]
```diff
@@ -42,7 +60,13 @@ def ShowKeys(self):
         return key_list
 
 
-# class ChanceOption(Option):
+class ChanceOption(Option):
```
Comment (agamm) [inline temp_library/classes.py]: ChanceOptions

**Review by agamm — CHANGES_REQUESTED** [review]
Read about the .gitignore file. Remove all the binary data (.pyc) and other code settings related files.
When you are done with all of the fixes ask me to review again.
Good job man!

File: temp_library/classes.py (hunk 11 above)
Comment (arielofer) [reply temp_library/classes.py]: i do use pylint but it did not say anything about the CamelCase

Comment (arielofer) [reply temp_library/classes.py]: i will change it anyway

**Comment by arielofer** [comment]
i noticed that .gitignore did not include .vscode and .pystest
how do i fix that?

**Comment by arielofer** [comment]
never mind. i fixed it with **git rm**

**Comment by agamm** [comment]
Resolve all of the things that you fixed, and click the button that asks me to re-review :+1:

File: example.py [hunk 13]
```diff
@@ -7,6 +7,7 @@
 def main():
```
Comment (agamm) [inline example.py:6]: We will go through this file again sometime when we fix all of the classes - remind me.

File: temp_library/chance_option.py [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import random
+from temp_library.option import *
+
+class ChanceOptions(Option):
+    
+    def DecisionDraw(self):
```
Comment (agamm) [inline temp_library/chance_option.py]: Please: https://www.python.org/dev/peps/pep-0257/

Comment (agamm) [reply temp_library/chance_option.py]: And other main classes functions.

File: temp_library/decision.py [hunk 15]
```diff
@@ -0,0 +1,14 @@
+class Decision(object):
```
Comment (agamm) [inline temp_library/decision.py]: Awesome. Next pull request I want tests for each class, ok?

File: temp_library/helpers.py [hunk 16]
```diff
@@ -3,117 +3,103 @@
 
 
 def openscrn():  # for my personal game-change to your liking
```
Comment (agamm) [inline temp_library/helpers.py]: A helper function can not have a specific value. Pass it through as a parameter. Add the specific code in the example.

File: temp_library/helpers.py [hunk 17]
```diff
@@ -3,117 +3,103 @@
...
+
 
 
 def statbuild():  # inserts initial values to attributes-change the strings and values according to your specific traits
     """stats setting function
 
     :rtype: Trait
     """
-
     trait1 = temp_library.trait.Trait("Skill", 6, 1)
```
Comment (agamm) [inline temp_library/helpers.py]: What is `temp_library`? change the name?

File: temp_library/helpers.py [hunk 18]
```diff
@@ -3,117 +3,103 @@
...
     trait2 = temp_library.trait.Trait("Luck", 6, 1)
     trait3 = temp_library.trait.Trait("Stamina", 12, 2)
 
     # trait1
-    print("to set your "+trait1.getName()+", please role "+str(trait1.getDiceNum())+" die and add "+str(trait1.getConst()))
+    print("to set your "+trait1.get_name()+", please role "+str(trait1.get_dice_num())+" die and add "+str(trait1.get_const()))
```
Comment (agamm) [inline temp_library/helpers.py]: We will have to work on this code. Right now leave it as is. But start thinking about how you would change this, and what exactly can be improved, ok?

File: temp_library/helpers.py [hunk 19]
```diff
@@ -3,117 +3,103 @@
...
     # displaying the stats function
     print(' ------------------------------')
-    print('|'+trait1.getName()+': ' + str(trait1.getValue()) + '|'+trait2.getName()+': '+str(trait2.getValue())+'|'+trait3.getName()+': '+str(trait3.getValue())+'|')
+    print('|'+trait1.get_name()+': ' + str(trait1.get_value()) + '|'+trait2.get_name()+': '+str(trait2.get_value())+'|'+trait3.get_name()+': '+str(trait3.get_value())+'|')
     print(' ------------------------------')
     print()
 
 
-def roll(num, arg):
+def roll(trait, arg):
```
Comment (agamm) [inline temp_library/helpers.py]: Have a better docstring, and example. It isn't really explainable by itself as it.

File: temp_library/helpers.py [hunk 20]
```diff
@@ -3,117 +3,103 @@
...
+
+    lucky_message = "success - press enter to continue..."
+    unlucky_message = "you were unlucky this time. press enter to continue..."
+
+    for i in range(dice_num):
+        dice_roll += random.randint(1, 6)
+
+    output = "you rolled a " + str(dice_roll)
 
     if arg == 'less':
```
Comment (agamm) [inline temp_library/helpers.py]: What is less?

File: temp_library/helpers.py [hunk 21]
```diff
@@ -3,117 +3,103 @@
...
 
 
 def fight(skill, stamina, m_skill, m_stamina):
     # fighting function
     print("your stamina "+str(stamina)+"monster stamina "+str(m_stamina))
-    Tskill = skill + random.randint(1, 6)
-    print("your skill is "+str(Tskill))
-    Tm_skill = m_skill + random.randint(1, 6)
-    print("monster skill is " + str(Tm_skill))
+    temp_skill = skill + random.randint(1, 6)
```
Comment (agamm) [inline temp_library/helpers.py]: Start thinking how you would implement this as classes (monster->defend? hero->attack? stuff like that - maybe you will have a better idea too :) )

File: temp_library/option.py [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+    def __init__(self, text, key, decision):
+        self.text = text
+        self.key = key
+        self.decision = decision
+
+    def show_text(self):
+        return self.text
+
+    def get_decision(self):
+        return self.decision.get_decision_name()
```
Comment (agamm) [inline temp_library/option.py]: Where does `decision` implement `get_decision_name`?

File: temp_library/option.py [hunk 23]
```diff
@@ -0,0 +1,21 @@
+class Option(object):
+    def __init__(self, text, key, decision):
```
Comment (agamm) [inline temp_library/option.py]: Add a small comment explaining this class (and others).

File: temp_library/scene.py [hunk 24]
```diff
@@ -0,0 +1,20 @@
...
+    def get_name(self):
+        return self.name
+
+    def show_desc(self):
+        return self.desc
+
+    def show_choices(self):
+        choice_list = "available options:\n"
+        for options in range(len(self.choices)-1):
+            choice_list += self.choices[options].show_text()+" "+self.choices[options].show_keys()+" "+\
```
Comment (agamm) [inline temp_library/scene.py]: If you are using python 3 please use string formatting, for example:
```
var='World'
a=f'Hello {var}'
print (a);
```

File: temp_library/scene.py [hunk 25]
```diff
@@ -0,0 +1,20 @@
+class Scene(object):
+    def __init__(self, name, desc, choices):
+        self.name = name
+        self.desc = desc
+        self.choices = choices
+        
+    def get_name(self):
```
Comment (agamm) [inline temp_library/scene.py]: Where are you using this?

File: temp_library/scene_collection.py [hunk 26]
```diff
@@ -0,0 +1,81 @@
+from temp_library.scene import *
+from temp_library.option import *
+from temp_library.decision import *
+from temp_library.helpers import openscrn
+from temp_library.chance_option import ChanceOptions
+
+
+class FleeScene(Scene):
```
Comment (agamm) [inline temp_library/scene_collection.py]: Put this file in an example directory + including all other files that are examples. No need to interfere with the main source code of the library.

File: temp_library/trait.py [hunk 27]
```diff
@@ -1,10 +1,9 @@
-"""skeleton for TextAdventure package"""
 import random
 
 
 class Trait(object):
-    def __init__(self, name, const, dice_num):
-        """rolled based trait
+    def __init__(self, name, const, dice_num=0):
+        """
             name: trait's name
```
Comment (agamm) [inline temp_library/trait.py]: Please don't explain things with the same word you are trying to define. Also, don't explain things that are very easy to grasp from the code itself. I would prefer something more related to what is a Trait, why is there this class in the first place 😄

**Review by agamm — CHANGES_REQUESTED** [review]
Looks better than before. See the changes. 
Next time Let's make PR's smaller so I could go through it faster and more specifically, ok?
After you fix all of the comments merge it, and the next PR do tests for each class, then we will continue.

**Comment by agamm** [comment]
בלי קשר בכלל, אחלה סרטון שמצאתי שחשבתי שיהיה נחמד לך
https://missing.csail.mit.edu/2020/potpourri/

**Comment by agamm** [comment]
@arielofer תעשה   resolve 
לכל מה שתיקנת, אם יש משהו שאתה לא בטוח תגיב ונפתור ביחד 👍

**Comment by arielofer** [comment]
האמת שלאחרונה היה קשה לי לפנות זמן לפרויקט אני משתדל לשבת על זה כמה שאני יכול.

File: example.py (hunk 13 above)
Comment (agamm) [reply example.py:6]: You can create a new example with only the new syntax.

File: temp_library/chance_option.py (hunk 14 above)
Comment (agamm) [reply temp_library/chance_option.py]: """
DESCRIPTION
"""

**Comment by arielofer** [comment]
@agamm 
.הוספתי את כול הדוקסטרינגים
?merge לעשות

**Comment by agamm** [comment]
> @agamm
> .הוספתי את כול הדוקסטרינגים
> ?merge לעשות

Yep,  נעשה code review  
להכל אחרי שנוסיף טסטים וכאלה.
יאללה הבא זה טסטים.

**Pull request merged**